client/fs_endpoint.go

```diff
@@ -588,11 +590,11 @@ func (f *FileSystem) logsImpl(ctx context.Context, follow, plain bool, offset in
 			close(eofCancelCh)
 			exitAfter = true
 		} else {
-			eofCancelCh = blockUntilNextLog(ctx, fs, logPath, task, logType, idx+1)
+			newFileChannelCh = blockUntilNextLog(ctx, fs, logPath, task, logType, idx+1)
```

Comment:
We've swapped out the `eofCancelCh` for the `newFileChannelCh` here, but now as far as I can tell there's nothing that'll ever send on `eofCancelCh`?

Reply:
I think it could be sent. When the follow flag is set to false then a closed channel is passed to end the loop when eof is reached

Reply:
Ah yeah I see it now. But given that it will _always_ be closed if it's non-nil, does it really need to be a channel at all now? We could pass a bool that sets the `cancelReceived`, so that as soon as we hit a EOF we return without having to set up the watcher. (That would let us further clean up the channel name too, because it becomes `eofCancelCh` again without confusion.)

Reply:
Sorry for the delay. I agree with you and applied this comment. 

We could even avoid adding a separate flag  by making an assumption that if `eofCancelCh` is closed then we return with additional read and perform one more when on standard event. But current way with a boolean is more explicit and easier to read probably.

Reply:
No worries on the delay... we've all been off for the end-of-year anyways 😀